src/__tests__/non-det-interpreter.ts

```diff
@@ -208,32 +208,130 @@ test('Require operator', async () => {
   )
 })
 
+/*  Deterministic block scoping tests taken from block-scoping.ts */
+
+test('Block statements', async () => {
+  await testDeterministicCode(
+    `
+    function test(){
+      const x = true;
+      {
+          const x = false;
+      }
+      return x;
+    }
+    test();
+    `,
+    true
+  )
+
+  await testDeterministicCode(
+    `
+    function test(){
+      let x = true;
+      if(true) {
+          let x = false;
+      } else {
+          let x = false;
+      }
+      return x;
+    }
+    test();
+    `,
+    true
+  )
+
+  await testDeterministicCode(
+    `
+    const v = f();
+    function f() {
+      return 1;
+    }
+    v;
+    `,
+    'Line -1: Name f declared later in current scope but not yet assigned',
+    true
+  )
+
+  await testDeterministicCode(
+    `
+    const a = 1;
+    function f() {
+      display(a);
+      const a = 5;
+    }
+    f();
+    `,
+    'Line -1: Name a declared later in current scope but not yet assigned',
```

Comment:
Why is this Line -1 ?

Reply:
See #29 - I'm not sure of the reason yet.

Reply:
Thanks for creating the issue